Projects/Core/Services/UIApplication/UIApplicationClient.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  UIApplicationClient.swift
+//  Services
+//
+//  Created by Hyun A Song on 7/6/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import ComposableArchitecture
+import UIKit
+
+@DependencyClient
+public struct UIApplicationClient: Sendable {
+  public var openURL: @Sendable () async throws -> Void
+}
+
+extension UIApplicationClient: DependencyKey {
+  public static var liveValue: UIApplicationClient {
+    return UIApplicationClient(
+      openURL: { @MainActor in
+        try await withCheckedThrowingContinuation { continuation in
+          if let url = URL(string: UIApplication.openSettingsURLString) {
+            UIApplication.shared.open(url)
```

Comment:
```suggestion
            UIApplication.shared.open(url) { success in
              if success {
               continuation.resume(returning: ())
              } else {
                continuation.resume(throwing: UIApplicationClientError(code: .failToGetUrl))
              }
            }
```

open 시, completionHandler 인자가 존재할텐데 그렇다면 open에 대해 성공, 실패가 존재한다는 의미일것 같아요.
성공 실패에 대해서도 처리해놓는게 어떨까요?

Reply:
와 놓쳤던 부분입니다..! Bool값을 반환하여 실패 시 ToastView를 띄우도록 하겠습니다!